ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -2147,7 +2151,8 @@ protected synchronized final int prepare (boolean reportHeuristics)
 
             ActionManager.manager().remove(get_uid());
 
-            return TwoPhaseOutcome.PREPARE_ONE_PHASE_COMMITTED;
+            return actionStatus == ActionStatus.ABORTED
+                    ? TwoPhaseOutcome.ONE_PHASE_ERROR : TwoPhaseOutcome.PREPARE_ONE_PHASE_COMMITTED;
```

Comment:
I disagree since the subclass can call BasicAction#status () to determine the result of calling onePhaseCommit